src/components/AddExisitingStudentsModal.tsx

```diff
@@ -47,12 +42,16 @@ const AddExistingStudentsModal: React.FC<AddExistingStudentsModalProps> = ({
     selectedStudentsToAdd.forEach(student => {
       const newStudentInStudentGroupRef = studentsInStudentGroupsRef.doc()
       addBatch.set(newStudentInStudentGroupRef, {
-        studentName: student.studentName,
+        studentInfo: {
+          studentName: student.studentName,
+          profilePic: student.profilePic,
+          favoriteFood: student.favoriteFood,
+        },
         studentId: student.studentId,
         studentGroupId: studentGroupDocument.docId,
         studentGroupName: studentGroupDocument.studentGroupName,
         selected: false,
-        order: studentsInThisStudentGroupLength++,
+        order: ++studentsInThisStudentGroupLength,
```

Comment:
why prepend instead of postpend?

Reply:
I want the first value to be one more than the length, since I started with 1 and not 0 for the order property on each studentInStudentGroup.  So if I have 10 students already, I want the new names to start being numbered at 11. Does that seem reasonable?

Reply:
would it have been the right thing to do to zero index the ordering instead of starting with 1?  at a minimum this is the kind of thing i would comment because it's pretty abstruse as to why you need to do it and it violates the principle of least astonishment.

Reply:
Okay, it's easy enough to switch it, and if it seems more confusing I might as well go ahead and do it

Reply:
I probably should have put this latest commit on a different PR, so sorry about that.  I went to zero index the ordering function, and I realized how to fix a related problem I was having with the names being added inconsistently.  I decided I wanted the names to be added to the beginning of the list always, so I modified the function that adds Students to StudentGroups to search for the lowest existing order on a StudentInStudentGroup and then give new students an order that's lower than whatever that starting number is.  This makes it easy to see when students are added, which gives the user that nice feedback that things are working. I did prepend -- though, because it seemed to make sense in this case as it wasn't just correcting my choice to not zero index.  Is prepending considered too obscure for most people to feel comfortable with it?  Is it better to do something like this (relevent lines are bold)?

```
const addExistingHandler = () => {
    let lowestOrderNumber = 0
    studentsInThisStudentGroupDocuments.forEach(student => {
      if (student.order < lowestOrderNumber) {
        lowestOrderNumber = student.order
      }
    })
    selectedStudentsToAdd.forEach(student => {
      const newStudentInStudentGroupRef = studentsInStudentGroupsRef.doc()
      addBatch.set(newStudentInStudentGroupRef, {
        studentInfo: {
          studentName: student.studentName,
          profilePic: student.profilePic,
          favoriteFood: student.favoriteFood,
        },
        studentId: student.studentId,
        studentGroupId: studentGroupDocument.docId,
        studentGroupName: studentGroupDocument.studentGroupName,
        selected: false,
        **order: lowestOrderNumber - 1,**
      })
        **lowestOrderNumber--**
    })
    return addBatch
      .commit()
      .then(() => {
        onClose()
        setSelectedStudentsToAdd([])
      })
      .catch(err => console.log(err))
  }
```

Reply:
well, I thought they'd be bold anyway.  Look for asterisks, lol